scopes/angular/ng-packagr/ng-packagr.compiler.ts

```diff
@@ -0,0 +1,160 @@
+          es2015: join(this.distDir, tempPackageJson.es2015),
+          esm2015: join(this.distDir, tempPackageJson.esm2015),
+          fesm2015: join(this.distDir, tempPackageJson.fesm2015),
+          typings: join(this.distDir, tempPackageJson.typings),
+          metadata: join(this.distDir, tempPackageJson.metadata),
+          sideEffects: tempPackageJson.sideEffects === 'true',
+          ngPackage: undefined,
+        });
+        await packageJson.write();
+        // delete the package.json file generated by ngPackagr
+        await removeFilesAndEmptyDirsRecursively([resolve(join(pathToOutputFolder, 'dist', 'package.json'))]);
+      })
+      .finally(() => {
+        this.logger.on();
+      });
+  }
+
+  /**
+   * used by `bit compile`
+   */
+  async transpileComponent(component: ConsumerComponent | Component, params: TranspileComponentParams): Promise<void> {
+    return this.ngPackagrCompilation(params.componentDir, params.distDir);
+  }
+
+  private getArtifactDefinition() {
+    return [
+      {
+        generatedBy: this.id,
+        name: this.artifactName,
+        globPatterns: this.distGlobPatterns,
+      },
+    ];
+  }
+
+  /**
+   * used by `bit build`
+   */
+  async build(context: BuildContext): Promise<BuiltTaskResult> {
+    const capsules = context.capsuleNetwork.seedersCapsules;
+    const componentsResults: ComponentResult[] = [];
+
+    await Promise.all(
+      context.components.map(async (component: Component) => {
+        const capsule = capsules.getCapsule(component.id);
+        if (!capsule) {
+          throw new Error(`No capsule found for ${component.id} in network graph`);
+        }
+        const currentComponentResult: ComponentResult = {
+          component,
+        };
+        try {
+          await this.ngPackagrCompilation(capsule.path, capsule.path);
+        } catch (e) {
+          currentComponentResult.errors = [e];
+        }
+
+        if (this.shouldCopyNonSupportedFiles) {
+          const dataToPersist = new DataToPersist();
+          capsule.component.filesystem.files.forEach((file: AbstractVinyl) => {
+            if (!this.isFileSupported(file.path)) {
+              dataToPersist.addFile(file);
+            }
+          });
+          dataToPersist.addBasePath(this.workspace.path);
+          await dataToPersist.persistAllToFS();
+        }
+
+        componentsResults.push({ ...currentComponentResult });
+      })
+    );
+
+    return {
+      artifacts: this.getArtifactDefinition(),
+      componentsResults,
+    };
+  }
+
+  /**
+   * given a source file, return its parallel in the dists. e.g. index.ts => dist/index.js
+   */
+  getDistPathBySrcPath(srcPath: string): string {
+    const fileWithJSExtIfNeeded = this.replaceFileExtToJs(srcPath);
+    return join(this.distDir, fileWithJSExtIfNeeded);
+  }
+
+  /**
+   * whether ngPackagr is able to compile the given path
+   */
+  isFileSupported(filePath: string): boolean {
+    return filePath.endsWith('.ts');
```

Comment:
js / jsx / tsx are not supported by definition?

Reply:
jsx/tsx I don't think so, js should work as long as the TS option is enabled, I'll test it